tests/test_unit/test_raw_to_ready.py

```diff
@@ -0,0 +1,253 @@
+            "threshold_method": "triangle",
+            "closing_size": 5,
+            "erode_size": 0,
+        },
+        "pad_pixels": 5,
+    }
+
+    config_path = path / "config.yaml"
+    with open(config_path, "w") as outfile:
+        yaml.dump(yaml_dict, outfile, default_flow_style=False)
+
+    return config_path
+
+
+@pytest.fixture()
+def create_raw_test_data(
+    tmp_path: Path, test_data: list[dict[str, Any]]
+) -> tuple[Path, Path]:
+    """Sets up temp directory with "raw" test images, CSV, and config files."""
+    raw_dir = tmp_path / "raw"
+    raw_dir.mkdir(parents=True, exist_ok=True)
+
+    test_data = create_test_images(raw_dir, test_data)
+    csv_path = create_test_csv(raw_dir, test_data)
+    config_path = create_test_yaml(raw_dir)
+    return csv_path, config_path
+
+
+def test_raw_to_ready(create_raw_test_data: tuple[Path, Path]) -> None:
+    """Test that raw_to_ready creates expected derivatives directory."""
+    csv_path, config_path = create_raw_test_data
+    raw_to_ready(csv_path, config_path)
+    assert (create_raw_test_data[0].parents[0] / "derivatives").exists()
+
+
+def test_create_subject_dir(tmp_path: Path) -> None:
+    """Test that _create_subject_dir creates correct directory structure."""
+    sub_id = "test123"
+    sub_dir = _create_subject_dir(sub_id, tmp_path)
+    assert sub_dir.exists()
+    assert sub_dir == tmp_path / "derivatives" / f"sub-{sub_id}"
+
+
+def test_create_subject_dir_exists(tmp_path: Path) -> None:
+    """Test exist_ok=True for _create_subject_dir."""
+    sub_id = "test123"
+    _create_subject_dir(sub_id, tmp_path)
+    _create_subject_dir(sub_id, tmp_path)
+
+
+def test_save_niftis(
+    tmp_path: Path, test_stacks: dict[str, NDArray[np.float64]]
+) -> None:
+    """Test that _save_niftis saves both standard and flipped images."""
+    voxel_sizes = [1.0, 1.0, 1.0]
+    image_name = "test_image"
+
+    image_path, flipped_path = _save_niftis(
+        test_stacks["image"], voxel_sizes, tmp_path, image_name
+    )
+
+    assert image_path.exists() & flipped_path.exists()
+    assert image_path.name == f"{image_name}.nii.gz"
+    assert flipped_path.name == f"{image_name}_lrflip.nii.gz"
+
+
+@pytest.mark.parametrize(
+    ["image_type", "error"],
+    [
+        pytest.param("image", None, id="sym image: lr-flip == non-flip"),
+        pytest.param(
+            "image_asym", AssertionError, id="asym image: lr-flip != non-flip"
+        ),
+    ],
+)
+def test_save_niftis_lrflip(
+    image_type: str,
+    error: type[Exception] | None,
+    tmp_path: Path,
+    test_stacks: dict[str, NDArray[np.float64]],
+) -> None:
+    """Test lr-flipped asym images differ from non-flipped ones."""
+
+    voxel_sizes = [1.0, 1.0, 1.0]
+    image_name = "test_image"
+
+    image_path, flipped_path = _save_niftis(
+        test_stacks[image_type], voxel_sizes, tmp_path, image_name
+    )
+
+    image = load_any(image_path)
+    flipped_image = load_any(flipped_path)
```

Comment:
As well as testing that the flipped / non-flipped images aren't identical. You could also do something like:
```
np.testing.assert_equal(flipped_image, np.flip(image, axis=2))
```
to check that they are really flipped copies of the same image.